### PR TITLE
Feature/user settings

### PR DESCRIPTION
**Pull request by BandanaPandey**
**Why**

UserSettings DataStore is required to store user settings related information, like recently_visited_app, taskbar_position, desktop_wallpaper, desktop_app_shortcuts, latitude, longitude, last_login_at.

**Relations**

UserSetting will belongs to User and User has_one UserSettings.
There is one-to-one mapping between User and UserSetting

**Changes**

- Added schema and model for UserSetting
- Added association between User and UserSetting
- Added UserSetting Controller
- Added UserSetting Creation api
- Added User Show API(with UserSetting Details)
- Added TestCases for UserSettings: schema, model and controller

### CODE REVIEW

File: apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        |> put_status(200)
+        |> render("user_details.json", %{user_details: user})
+
+      404 ->
+        conn
+        |> send_error(404, "Resource Not Found")
+    end
+  end
+
+  defp load_user(%{params: %{"id" => user_id}} = conn, _params) do
```
Comment (arjun289) [inline apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex:23]: I will suggest you move this to `user_context`.

File: apps/acqdat_api/lib/acqdat_api_web/router.ex [hunk 2]
```diff
@@ -39,6 +39,10 @@ defmodule AcqdatApiWeb.Router do
     resources "/site", SiteController, only: [:create, :update, :delete, :index, :show]
     resources "/process", ProcessController, only: [:create, :update, :delete, :index, :show]
 
+    resources "/users", UserController, only: [:show] do
+      resources "/settings", UserSettingController, only: [:create]
```
Comment (arjun289) [inline apps/acqdat_api/lib/acqdat_api_web/router.ex:35]: This has only to create, but you mentioned we need to update also, let me know how is it handled.

File: apps/acqdat_core/lib/acqdat_core/schema/user_setting.ex [hunk 3]
```diff
@@ -0,0 +1,111 @@
+defmodule AcqdatCore.Schema.UserSetting do
+  @moduledoc """
+  Models a UserSetting in the system.
+
+  It will include all the user_settings related dat of users
```
Comment (arjun289) [inline apps/acqdat_core/lib/acqdat_core/schema/user_setting.ex:5]: type in `dat`

File: apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex (hunk 1 above)
Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex:23]: Ok, will do that.

File: apps/acqdat_core/lib/acqdat_core/schema/user_setting.ex (hunk 3 above)
Comment (BandanaPandey) [reply apps/acqdat_core/lib/acqdat_core/schema/user_setting.ex:5]: Ohh, will fix this typo

File: apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex (hunk 1 above)
Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex]: Yes, did that.

File: apps/acqdat_api/lib/acqdat_api_web/router.ex (hunk 2 above)
Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/router.ex:35]: @arjun289 , currently I haven't exposed the update API.
Should I expose it ?

Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/router.ex]: Added update API in Commit: fb48a6b980f7af95cbdecbc32515ab39eccba5c4, 66ab0253fa9cebbe1692c74993f6fb763c0d928f

File: apps/acqdat_core/lib/acqdat_core/schema/user_setting.ex (hunk 3 above)
Comment (BandanaPandey) [reply apps/acqdat_core/lib/acqdat_core/schema/user_setting.ex]: Fixed in Commit 28511f8c17cfe68ee2370c0328b92daa7b1944d4

File: apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex (hunk 1 above)
Comment (BandanaPandey) [reply apps/acqdat_api/lib/acqdat_api_web/controllers/user_controller.ex]: Fixed in e37837d6cfda2ce5c9f80eeda210c157b893a42e

**Pull request merged**